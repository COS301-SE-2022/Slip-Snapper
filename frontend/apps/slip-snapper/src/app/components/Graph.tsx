import { Bar } from "react-chartjs-2";

type Props = {
    graphData: any;
}
function Graph({ graphData }: Props) {

    const amounts = []
    const frequency = []
    for(let i = 0 ; i<graphData.amounts.length;i++)
    {
        amounts[i] = graphData.amounts[i].itemPrice
        frequency[i] = graphData.occurances[i].id
    }


     const graphSettings = {
        responsive: true,
        barThickness: 43,
        borderWidth: 1,
        borderRadius: 10,
        
        plugins: {
            legend: {
                position: 'bottom' as const,
            },
            title: {
                display: true,
                text: graphData.itemName+' prices across various stores',
            },
        },
    };

     const graphStats = {
         labels: graphData.stores,
        datasets: [
            {
                label: 'Price (R)',
                data: amounts,
                backgroundColor: "rgb(39, 165, 146, 0.8)",
                width: 8,
                hoverBackgroundColor: "rgb(47, 198, 176, 0.8)",
            },
            {
                label: 'Frequency',
                data: frequency,
                backgroundColor: 'rgb(156, 39, 176, 0.8)',
                width: 8,
                hoverBackgroundColor: "rgb(187, 54, 211, 0.8)",
                hidden: true,
            }
        ],
    };

    return (
        <Bar className="graph" options={graphSettings} data={graphStats} />
    );
}

export default Graph;