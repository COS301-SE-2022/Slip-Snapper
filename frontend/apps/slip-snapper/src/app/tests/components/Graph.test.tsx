import renderer from 'react-test-renderer';
import { Bar } from "react-chartjs-2";
import Graph from "../../components/Graph";

jest.mock("react-chartjs-2");

const renderTree = (tree: JSX.Element) => renderer.create(tree);
describe('Should render Graph component', () => {
  jest.spyOn(console, 'error').mockImplementation(() => jest.fn())
  it('should render component with props', () => {
    expect(renderTree(<Graph  
      graphData={{
        amounts:[{itemPrice:30}],
        itemName:"Coke",
        occurances:[ {id: 1}],
        stores:['Chicken Licken']
        }} 
    />).toJSON()).toMatchSnapshot();
  });
});