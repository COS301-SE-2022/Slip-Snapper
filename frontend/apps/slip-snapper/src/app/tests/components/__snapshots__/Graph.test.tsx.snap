// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Graph component should render component with props 1`] = `null`;
